src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ImportCompletionItem.cs

```diff
@@ -80,54 +97,48 @@ public static string GetContainingNamespace(CompletionItem item)
 
         public static async Task<CompletionDescription> GetCompletionDescriptionAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
         {
-            var metadataName = GetMetadataName(item);
-            if (!string.IsNullOrEmpty(metadataName))
+            var compilation = (await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false))!;
+            var symbol = GetSymbol(item, compilation);
+
+            if (symbol != null)
             {
-                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
-                var symbol = compilation.GetTypeByMetadataName(metadataName);
-                if (symbol != null)
-                {
-                    var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
-
-                    // We choose not to display the number of "type overloads" for simplicity. 
-                    // Otherwise, we need additional logic to track internal and public visible
-                    // types separately, and cache both completion items.
-                    return await CommonCompletionUtilities.CreateDescriptionAsync(
-                        document.Project.Solution.Workspace,
-                        semanticModel,
-                        position: 0,
-                        symbol,
-                        overloadCount: 0,
-                        supportedPlatforms: null,
-                        cancellationToken).ConfigureAwait(false);
-                }
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                // We choose not to display the number of "type overloads" for simplicity. 
+                // Otherwise, we need additional logic to track internal and public visible
+                // types separately, and cache both completion items.
+                return await CommonCompletionUtilities.CreateDescriptionAsync(
+                    document.Project.Solution.Workspace,
+                    semanticModel,
+                    position: 0,
```

Comment:
position:0 seems bad... is that intentional?  why dont we actually use the right position (i.e  the locaiton the user brought completion up at)?  if you use position:0 won't all the description types be fully qualified, instead of properly qualified at the trigger point?

Reply:
I will admit that I don't really know what position does to the description created, just saw it was created this way somewhere

Reply:
do descriptions are created lazily on demand.  it involves going back to hte original source, binding symbols, getting doc comments, and rendering them as if we were at the point the user triggered completion.   we store the position so we can do that properly and so that quick info will show the right descrption based on where the user is.  so we should def try to preserve position if possible...

(though if it turns out to be really hard, we could consider not doing that).

Reply:
OK, the problem is we cache all the completion items created this way (and we removed our dependency on Span because of the same reason), so we are currently not able to get this info w/o changing CompletionSource and provider interface. We can consider doing this in a follow up PR.

Reply:
got it.  so you intentionally do not keep track of positions because you want to cache/reuse the completion items for *different* positions?  if that's correct, please doc this.